src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/SentencePurpose.kt

```diff
@@ -0,0 +1,44 @@
+package uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa
+
+import jakarta.persistence.Column
+import jakarta.persistence.Embeddable
+import jakarta.persistence.EmbeddedId
+import jakarta.persistence.Entity
+import jakarta.persistence.Table
+import org.hibernate.Hibernate
+import org.hibernate.annotations.Generated
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.helper.EntityOpen
+import java.io.Serializable
+import java.time.LocalDateTime
+
+@Embeddable
+class SentencePurposeId(
+  var orderId: Long,
+  var orderPartyCode: String, // always 'CRT' on prod
+  var purposeCode: String,
```

Comment:
this reference data (purposeCode) is awkward to map as part of the embedded key. Any create will validate the reference code.